src/main/java/duchess/util/TaskList.java

```diff
@@ -0,0 +1,127 @@
+package duchess.util;
+
+import duchess.tasks.Task;
+
+import java.util.ArrayList;
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    /**
+     * Create new empty TaskList object.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Create new Task List object with previous tasks already loaded.
+     *
+     * @param taskList ArrayList of Task objects.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Print list of tasks.
+     */
+    public void printTaskList() {
+        System.out.println("\t\tHere are the tasks in your list:");
+        for (int i = 1; i <= taskList.size(); i++) {
+            Task t = taskList.get(i - 1);
+            System.out.println("\t\t" + i + "." + t);
+        }
+    }
+
+    /**
+     * Retrieve ArrayList of Task objects.
+     *
+     * @return ArrayList of Task objects.
+     */
+    public ArrayList<Task> retrieveTaskList() {
+        return this.taskList;
+    }
+
+    /**
+     * Delete Task object in ArrayList of Task objects.
+     *
+     * @param id id of task in list to be deleted.
+     */
+    public void deleteTask(int id) throws IndexOutOfBoundsException{

```

Comment:
It seems for this function, you may have missed a white space after the `IndexOutOfBoundsException`

There appears to be one other location, in the `addToDoTest` method in `TaskListTest`, that is missing a white space.

Other than that, everything else seems great 👍